list-ops/example.lua

```diff
@@ -0,0 +1,26 @@
+local function reduce(xs, value, f)
+  for _, x in ipairs(xs) do
+    value = f(x, value)
+  end
+  return value
+end
+
+local function map(xs, f)
+  return reduce(xs, {}, function(x, acc)
+    table.insert(acc, f(x))
+    return acc
+  end)
+end
+
+local function filter(xs, pred)
+  return reduce(xs, {}, function(x, acc)
+    if pred(x) then table.insert(acc, x) end
+    return acc
+  end)
+end
+
```

Comment:
I didn't really do this because it's the solution I expect. I did it because I find it interesting that you can implement map and filter in terms of reduce. My hope is that anyone curious enough to either look at my submission or peek into xlua might also find it interesting :)